src/lib.rs

```diff
@@ -1393,7 +1405,7 @@ impl Build {
                                 // sure users always see all the compilation failures.
                                 has_made_progress.set(true);
 
-                                let _ = writeln!(stdout, "cargo:warning={}", err);
+                                self.print_warning(&err);
```

Comment:
Just want to point out that this now behaves slightly differently because it no longer goes through the BufWriter and instead writes to stdout directly.
I don't really like this. Are you okay with keeping this as:

```rust
if self.cargo_warnings {
    let _ = writeln!(stdout, "cargo:warning={}", err);
}
```

Reply:
Yes, I'm ok with keeping the if and writeln